apps/beeai-server/src/beeai_server/utils/managed_server_client.py

```diff
@@ -37,6 +39,21 @@ class ManagedServerParameters(BaseModel):
     endpoint: str = "/sse"
 
 
+def _kill_process(process: anyio.abc.Process):
+    import psutil
+
+    parent = psutil.Process(process.pid)
+    for child in parent.children(recursive=True):  # or parent.children() for recursive=False
```

Comment:
Not sure what the comment means, I think we always want to go recursive (providers can spawn more processes).

Reply:
copy-paste from stack overflow :D